src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_COUNT = 5;
+    private static final String COLOR_DEFAULT = "WHITE";
+    private static final int RADIUS_DEFAULT = 10;
+    private Random random = new Random();
+    private String randomColor;
+    private Figure figure;
+
+    private Circle getRandomCircle() {
+        randomColor = new ColorSupplier().getRandomColor();
+        double randomRadius = random.nextInt(50);
+        return new Circle(randomColor, randomRadius);
+    }
+
+    private Square getRandomSquare() {
+        randomColor = new ColorSupplier().getRandomColor();
+        double randomSide = random.nextInt(50);
+        return new Square(randomColor, randomSide);
+    }
+
+    private RightTriangle getRandomRightTriangle() {
+        randomColor = new ColorSupplier().getRandomColor();
+        double randomFirstLeg = random.nextInt(50);
+        double randomSecondLeg = random.nextInt(50);
+        return new RightTriangle(randomColor, randomFirstLeg, randomSecondLeg);
+    }
+
+    private Rectangle getRectangle() {
+        randomColor = new ColorSupplier().getRandomColor();
+        double randomFirstSide = random.nextInt(50);
+        double randomSecondSide = random.nextInt(50);
+        return new Rectangle(randomColor, randomFirstSide, randomSecondSide);
+    }
+
+    private IsoscelesTrapezoid getIsoscelesTrapezoid() {
+        randomColor = new ColorSupplier().getRandomColor();
+        double randomFirstSideTr = random.nextInt(50);
+        double randomSecondSideTr = random.nextInt(50);
+        double randomHeight = random.nextInt(50);
+        return new IsoscelesTrapezoid(randomColor, randomFirstSideTr,
+                randomSecondSideTr, randomHeight);
+    }
+
+    public Figure getRandomFigure() {
+        FigureSupplier figureSupplier = new FigureSupplier();
```

Comment:
remove this variable, you don't need it to call methods for getting of random figures

Reply:
method was removed